### PR TITLE
Mint kudos improvements

### PR DESCRIPTION
**Pull request by Lamperoyge**
## :pushpin: References

- **Wonder issue:**
- **Related pull-requests:**

## :cake: Checklist:

- [ ] I self-reviewed my changes
- [ ] My changes follow the style guide: https://creative-earth-33e.notion.site/TT-Wonder-Style-guide-4702a45133374148953bfcaf584120b7
- [ ] I tested my changes in Chrome

### CODE REVIEW

File: wondrous-app/utils/hooks.ts [hunk 1]
```diff
@@ -310,9 +310,15 @@ export const useNotifications = () => {
 export const useSteps = (defaultStep = 0) => {
   const [step, setStep] = useState(defaultStep);
 
-  const nextStep = () => setStep((prevState) => prevState + 1);
+  const nextStep = () => {
+    if (step === step + 1) return;
```
Comment (terryli0095) [inline wondrous-app/utils/hooks.ts:314]: why would this be?

File: wondrous-app/utils/hooks.ts [hunk 2]
```diff
@@ -310,9 +310,15 @@ export const useNotifications = () => {
...
+    setStep((prevState) => prevState + 1);
+  };
 
-  const prevStep = () => setStep((prevState) => prevState - 1);
+  const prevStep = () => {
+    if (step === step - 1) return;
```
Comment (terryli0095) [inline wondrous-app/utils/hooks.ts:319]: same here

File: wondrous-app/components/Common/TaskMint/index.tsx [hunk 3]
```diff
@@ -4,33 +4,43 @@ import { useMe } from 'components/Auth/withAuth';
...
-
   if (taskStatus !== TASK_STATUS_DONE || user?.id !== assigneeId) {
     return null;
   }
 
   const toggleModal = () => setIsModalOpen((prevState) => !prevState);
+
+  const handleMintButtonClick = () => (status === TaskMintStatus.COMPLETED ? setIsViewNft(true) : toggleModal());
+
+  if (process.env.NEXT_PUBLIC_PRODUCTION) return null;
```
Comment (terryli0095) [inline wondrous-app/components/Common/TaskMint/index.tsx:33]: this is the current feature flag right?

File: wondrous-app/components/Common/TaskMint/TaskMintViewActions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Grid } from '@mui/material';
+import { SnackbarAlertContext } from 'components/Common/SnackbarAlert';
+import { useContext } from 'react';
+import { LINK } from 'utils/constants';
+import OpenseaButton from '../OpenseaButton';
+import { CommunityShare } from '../Steps/SuccessMint';
+import { Wrapper } from './styles';
+
+const TaskViewMintActions = ({ tokenId, taskId }) => {
```
Comment (terryli0095) [inline wondrous-app/components/Common/TaskMint/TaskMintViewActions/index.tsx:9]: maybe rename it to MintViewAndShareOptions?
naming is hard here, perhaps just add a comment explaining which component

File: wondrous-app/components/Common/TaskMint/index.tsx (hunk 3 above)
Comment (Lamperoyge) [reply wondrous-app/components/Common/TaskMint/index.tsx:33]: yes, correct

**Pull request merged**